src/main/java/pl/put/poznan/sorting/logic/AutoDetector.java

```diff
@@ -1,15 +1,43 @@
 package pl.put.poznan.sorting.logic;
 
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+
+/**
+ * Class for detecting optimal algorithm for the specific dataset
+ */
 public class AutoDetector {
-    private int[] data = null;
+    private int[] array = null;
```

Comment:
To tylko propozycja, ale chyba lepszym rozwiązaniem byłoby nie rozbijanie `data` na dwa różne rodzaje zmiennych, tylko w atrybucie `type` przechowywać rezultat `getType` (bo tylko to ta klasa robi). Z poziomu konstruktora wywołaj `getType` (gdzie w `AutoDetector(int[])` dasz `getType('array')` a w `AutoDetector(ArrayList<>)` dasz `getType('object')`) i przypisz rezultat do atrybutu `type`. Wówczas nie musisz niczego rozbijać, potem możesz dodać gettera który będzie zwracał samą wartość `type` i ją zwracać w pozostałych klasach.